pkg/cvo/upgradeable.go

```diff
@@ -466,21 +492,20 @@ func (optr *Operator) adminGatesEventHandler() cache.ResourceEventHandler {
 	}
 }
 
-// shouldSyncUpgradeableDueToPreconditionChecks checks if the upgradeable status should
-// be synchronized due to the precondition checks. It checks whether the precondition
-// checks on the payload are failing for less than minimumUpdateCheckInterval, and it has
-// been more than the minimumUpgradeableCheckInterval since the last synchronization.
-// This means, upon precondition failure, we will synchronize the upgradeable status
-// more frequently at beginning of an upgrade.
-//
-// shouldSyncUpgradeableDueToPreconditionChecks expects the parameters not to be nil.
+// throttlePeriod returns the duration for which upgradeable status should be considered recent
+// enough and unnecessary to update. The baseline duration is min. When the precondition checks
+// on the payload are failing for less than afterPreconditionsFailed we want to synchronize
+// the upgradeable status more frequently at beginning of an upgrade and return
+// minOnFailedPreconditions which is expected to be lower than min.
 //
-// Function returns true if the synchronization should happen, returns false otherwise.
-func shouldSyncUpgradeableDueToPreconditionChecks(optr *Operator, config *configv1.ClusterVersion, u *upgradeable) bool {
-	cond := resourcemerge.FindOperatorStatusCondition(config.Status.Conditions, DesiredReleaseAccepted)
-	if cond != nil && cond.Reason == "PreconditionChecks" && cond.Status == configv1.ConditionFalse &&
-		hasPassedDurationSinceTime(u.At, optr.minimumUpgradeableCheckInterval) && !hasPassedDurationSinceTime(cond.LastTransitionTime.Time, optr.minimumUpdateCheckInterval) {
-		return true
-	}
-	return false
+// The cv parameter is expected to be non-nil.
+func (intervals *upgradeableCheckIntervals) throttlePeriod(cv *configv1.ClusterVersion) time.Duration {
+	if cond := resourcemerge.FindOperatorStatusCondition(cv.Status.Conditions, DesiredReleaseAccepted); cond != nil {
+		// Function returns true if the synchronization should happen, returns false otherwise.
```

Comment:
nit: this comment probably belongs to the previous `func shouldSyncUpgradeableDueToPreconditionChecks`

Reply:
good catch, I want to submit a little code cleanup PR for this area that wasnt appropriate for a to-be-backported bugfix, I'll make sure I fix this there

Reply:
Included in https://github.com/openshift/cluster-version-operator/pull/886